src/engine/Server.cpp

```diff
@@ -938,31 +927,35 @@ CPP_template_2(typename RequestT, typename ResponseT)(
         const ad_utility::Timer& requestTimer,
         ad_utility::SharedCancellationHandle cancellationHandle,
         QueryExecutionContext& qec, const RequestT& request, ResponseT&& send,
-        TimeLimit timeLimit) {
+        TimeLimit timeLimit, std::optional<PlannedQuery>& plannedUpdate) {
   AD_CORRECTNESS_CHECK(ql::ranges::all_of(
       updates, [](const ParsedQuery& p) { return p.hasUpdateClause(); }));
 
+  // Atomicity of the update is still given, because everything runs on
+  // `updateThreadPool_` which only has one thread.
```

Comment:
If multiple updates are part of a single request, those have to run atomically. This is ensured, because the updates below are run on the `updateThreadPool_`, which only has a single thread.